src/pages/home/report/ReportActionItemSingle.tsx

```diff
@@ -21,112 +18,136 @@ import DateUtils from '@libs/DateUtils';
 import Navigation from '@libs/Navigation/Navigation';
 import * as ReportUtils from '@libs/ReportUtils';
 import * as UserUtils from '@libs/UserUtils';
-import reportPropTypes from '@pages/reportPropTypes';
-import stylePropTypes from '@styles/stylePropTypes';
 import CONST from '@src/CONST';
 import ROUTES from '@src/ROUTES';
+import type {Report, ReportAction} from '@src/types/onyx';
+import {AvatarType} from '@src/types/onyx/OnyxCommon';
 import ReportActionItemDate from './ReportActionItemDate';
 import ReportActionItemFragment from './ReportActionItemFragment';
-import reportActionPropTypes from './reportActionPropTypes';
 
-const propTypes = {
+type ReportActionItemSingleProps = {
     /** All the data of the action */
-    action: PropTypes.shape(reportActionPropTypes).isRequired,
+    action: ReportAction;
 
     /** Styles for the outermost View */
-    wrapperStyle: stylePropTypes,
+    wrapperStyle?: StyleProp<ViewStyle>;
 
     /** Children view component for this action item */
-    children: PropTypes.node.isRequired,
+    children: React.ReactNode;
 
     /** Report for this action */
-    report: reportPropTypes,
+    report: Report;
 
     /** IOU Report for this action, if any */
-    iouReport: reportPropTypes,
+    iouReport?: Report;
 
     /** Show header for action */
-    showHeader: PropTypes.bool,
+    showHeader?: boolean;
 
     /** Determines if the avatar is displayed as a subscript (positioned lower than normal) */
-    shouldShowSubscriptAvatar: PropTypes.bool,
+    shouldShowSubscriptAvatar?: boolean;
 
     /** If the message has been flagged for moderation */
-    hasBeenFlagged: PropTypes.bool,
+    hasBeenFlagged?: boolean;
 
     /** If the action is being hovered */
-    isHovered: PropTypes.bool,
-
-    ...withLocalizePropTypes,
+    isHovered?: boolean;
 };
 
-const defaultProps = {
-    wrapperStyle: undefined,
-    showHeader: true,
-    shouldShowSubscriptAvatar: false,
-    hasBeenFlagged: false,
-    report: undefined,
-    iouReport: undefined,
-    isHovered: false,
+type SubAvatar = {
```

Comment:
Is this type not defined anywhere yet?

Reply:
Actually there is 😅 

`Icon` from `src/types/onyx/OnyxCommon.ts`